public/nextjs_migration/client/js/breaches.js

```diff
@@ -0,0 +1,223 @@
+  } finally {
+    input.disabled = false
+  }
+}
+
+function renderResolvedCounts () {
+  state.resolvedCount = breachesPartial.querySelectorAll(`[data-status='resolved'][data-email='${state.selectedEmail}']`).length
+  state.unresolvedCount = breachesPartial.querySelectorAll(`[data-status='unresolved'][data-email='${state.selectedEmail}']`).length
+  resolvedCountOutput.textContent = state.resolvedCount
+  unresolvedCountOutput.textContent = state.unresolvedCount
+}
+
+function renderBreachRows () {
+  let delay = 0
+  let hidden
+
+  breachRows.forEach(breach => {
+    hidden = (breach.dataset.email !== state.selectedEmail) || (breach.dataset.status !== state.selectedStatus)
+    breach.toggleAttribute('hidden', hidden)
+    breach.removeAttribute('open')
+    if (!hidden) {
+      breach.style.setProperty('--delay', `${delay}ms`)
+      delay += 50
+    }
+  })
+}
+
+function renderZeroState () {
+  let temp
+
+  breachesTable.querySelector('.zero-state')?.remove()
+  statusFilter.toggleAttribute('disabled', state.resolvedCount === 0 && state.unresolvedCount === 0)
+
+  switch (true) {
+    case state.resolvedCount === 0 && state.unresolvedCount === 0:
+      temp = breachesPartial.querySelector('template.no-breaches')
+      break
+    case state.resolvedCount > 0 && state.unresolvedCount === 0:
+      if (state.selectedStatus !== 'unresolved') return // only show zero-state on empty unresolved screen
+      temp = breachesPartial.querySelector('template.all-breaches-resolved')
+      break
+    default:
+      return
+  }
+
+  const content = temp.content.cloneNode(true)
+  content.querySelector('.current-email').textContent = state.selectedEmail
+  content.querySelector('.add-email-cta').toggleAttribute('hidden', state.emailCount >= state.emailTotal)
+  breachesTable.append(content)
+}
+
+/**
+ * Capitalize the first letter of a string, taking into account localization
+ *
+ * @param {string} str
+ * @example
+ * capitalFirstLetter ('carte di credito')
+ * // returns 'Carte di credito'
+ * @example
+ * capitalFirstLetter ('账户余额')
+ * // returns '账户余额'
+ */
+
+function capitalFirstLetter (str) {
+  return str[0].toLocaleUpperCase() + str.slice(1)
+}
+
+function renderPieChart () {
+  const rowsForSelectedEmail = Array.from(breachesTable.querySelectorAll(`[data-email='${state.selectedEmail}']`))
+  const classesForSelectedEmail = rowsForSelectedEmail.flatMap(row => row.dataset.classes.split(','))
+  const classesMap = classesForSelectedEmail.reduce((acc, cur) => {
+    acc.set(cur, (acc.get(cur) ?? 0) + 1) // set count for each class key
+    return acc
+  }, new Map())
+  const classesTop3 = [...classesMap.keys()].sort((a, b) => classesMap.get(b) - classesMap.get(a)).slice(0, 3)
+  const classesTotal = classesForSelectedEmail.length
+  const chartData = []
+
+  switch (true) {
+    case classesMap.size === 0:
+      chartData.push({
+        key: pieChart.dataset.txtNone,
+        name: capitalFirstLetter(pieChart.dataset.txtNone),
```

Comment:
Q: Why do we need `capitalFirstLetter()` aren't the strings already translated in Pontoon?

Reply:
Good point — we should not alter translated strings. Removed: [8f31321](https://github.com/mozilla/blurts-server/pull/3085/commits/8f3132170cdfcb369eafe4f076cdf0b33dbd35b8).